### PR TITLE
Initial commit of broker channel actor implementation

### PR DESCRIPTION
**Pull request by autodidaddict**
Not yet implemented: Interactions with the `auth` and `presence` actors. Right now this is just the most basic interactions and I am checking this into a draft PR prior to heading out for the rest of the day.

### CODE REVIEW

File: wasmcloud-chat/actors/broker-channel/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+fn handle_outbound_message(msg: messaging::BrokerMessage) -> HandlerResult<()> {
+    let cloud_event: CloudEvent = serde_json::from_slice(&msg.body)?;
+    let subject = msg
+        .subject
+        .replace(BACKEND_SUBJECT_PREFIX, FRONTEND_SUBJECT_PREFIX);
+    let _ = messaging::host(LINK_NAME_FRONTEND).publish(
+        subject,
+        "".to_string(),
+        serde_json::to_vec(&cloud_event)?,
+    )?;
```
Comment (brooksmtownsend) [inline wasmcloud-chat/actors/broker-channel/src/lib.rs:92]: So this explicit deserialize/serialize is to make sure that the data isn't just blindly forwarded, it's asserted to be a CloudEvent? That's a neat pattern.

**Review by brooksmtownsend — APPROVED** [review]

File: wasmcloud-chat/actors/broker-channel/src/lib.rs (hunk 1 above)
Comment (autodidaddict) [reply wasmcloud-chat/actors/broker-channel/src/lib.rs:92]: Exactly.

By unraveling here we also now have a good spot where we can add augmentation / decoration to the cloud event during proxying if we choose.

File: wasmcloud-chat/ops/broker-message-request.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{ "source_app_id": "website", "source_app_version": "1.0", "message_text": "Hello from the other side", "origin_user_id": "bob", "target_url": "wcchat://rooms/general", "created_on": 0, "access_token": "" }
```
Comment (brooksmtownsend) [inline wasmcloud-chat/ops/broker-message-request.json:1]: I believe `origin_user_id` should be `adele` here.

Comment (autodidaddict) [reply wasmcloud-chat/ops/broker-message-request.json:1]: Indeed. :monocle_face:

**Pull request merged**